wrappers/kn/src/nativeMain/kotlin/zxingcpp/ImageView.kt

```diff
@@ -57,3 +59,32 @@ enum class ImageFormat(internal val cValue: ZXing_ImageFormat) {
 @OptIn(ExperimentalForeignApi::class)
 fun ZXing_ImageFormat.parseIntoImageFormat(): ImageFormat? =
 	ImageFormat.entries.firstOrNull { it.cValue == this }
+
+@ExperimentalWriterApi
+@OptIn(ExperimentalForeignApi::class)
+class Image(val cValue: CValuesRef<ZXing_Image>) {
+	val data: ByteArray
+		get() = ZXing_Image_data(cValue)?.run {
+			readBytes(width * height).also { ZXing_free(this) }
+		}?.takeUnless { it.isEmpty() } ?: throw OutOfMemoryError()
+	val width: Int get() = ZXing_Image_width(cValue)
+	val height: Int get() = ZXing_Image_height(cValue)
+	val format: ImageFormat
+		get() = ZXing_Image_format(cValue).parseIntoImageFormat() ?: error(
+			"Unknown format ${
```

Comment:
This is actually an "internal error" I believe, because there is no user in put that can ever result in this circustance, only a bug in the library implementation.

Reply:
I marked in up rather than remove for it reduce users' effort on debug if there's really something wrong with the implementation: https://github.com/zxing-cpp/zxing-cpp/pull/764/commits/7c4afeafab364428061fc8c0d9b0c61b7c18bfc9